### ISSUE TITLE
[🐛 Bug]: npm run test fails when creating WebdriverIO development environment on Windows 10

### ISSUE THREAD

**Issue by RossVertizan** [webdriverio/webdriverio#9078]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

latest

### Node.js Version

v18.11.0

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
Not related to capabilities
```

### What happened?

When creating a WebdriverIO development environment on Windows 10 and following the steps in the documentation the test fails.
```shell
git clone https://github.com/webdriverio/webdriverio.git webdriverio_1
cd webdriverio_1
npm install
npm run setup-full
npm run test
```
fails with this output:
```shell
PS H:\Git\webdriverio_1> npm run test

> test
> run-s test:eslint test:typings test:unit test:smoke


> test:eslint
> eslint --cache packages examples scripts tests


> test:typings
> run-s test:typings:*


> test:typings:setup
> node tests/typings/setup && npm i --no-save tsd

ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist

added 15 packages, and audited 1210 packages in 9s

181 packages are looking for funding
  run `npm fund` for details

4 vulnerabilities (2 high, 2 critical)

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

> test:typings:test
> npm run ts && npm run clean:tests && node tests/typings/setup --ts=3.9 && npm run ts && npm run clean:tests


> ts
> run-p ts:*


> ts:webdriverio
> cd tests/typings/webdriverio && npx tsc --skipLibCheck


> ts:devtools
> cd tests/typings/devtools && npx tsc --skipLibCheck


> ts:jasmine
> cd tests/typings/jasmine && npx tsc --skipLibCheck


> ts:cucumber
> cd tests/typings/cucumber && npx tsc --skipLibCheck


> ts:webdriver
> cd tests/typings/webdriver && npx tsc --skipLibCheck


> ts:mocha
> cd tests/typings/mocha && npx tsc --skipLibCheck


> clean:tests
> rimraf tests/typings/**/node_modules tests/typings/**/dist ./coverage

ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist

> ts
> run-p ts:*


> ts:webdriver
> cd tests/typings/webdriver && npx tsc --skipLibCheck


> ts:devtools
> cd tests/typings/devtools && npx tsc --skipLibCheck


> ts:mocha
> cd tests/typings/mocha && npx tsc --skipLibCheck


> ts:cucumber
> cd tests/typings/cucumber && npx tsc --skipLibCheck


> ts:webdriverio
> cd tests/typings/webdriverio && npx tsc --skipLibCheck


> ts:jasmine
> cd tests/typings/jasmine && npx tsc --skipLibCheck


> clean:tests
> rimraf tests/typings/**/node_modules tests/typings/**/dist ./coverage


> test:typings:recover
> rimraf ./node_modules/.bin/tsc && ln -s ../typescript/bin/tsc node_modules/.bin && npx rimraf @tsd/typescript

'ln' is not recognized as an internal or external command,
operable program or batch file.
ERROR: "test:typings:recover" exited with 1.
ERROR: "test:typings" exited with 1.
```

### What is your expected behavior?

That the tests pass.

### How to reproduce the bug.

On Windows 10 run the following commands:
```shell
git clone https://github.com/webdriverio/webdriverio.git webdriverio_1
cd webdriverio_1
npm install
npm run setup-full
npm run test
```

### Relevant log output

```typescript
PS H:\Git\webdriverio_1> npm run test

> test
> run-s test:eslint test:typings test:unit test:smoke


> test:eslint
> eslint --cache packages examples scripts tests


> test:typings
> run-s test:typings:*


> test:typings:setup
> node tests/typings/setup && npm i --no-save tsd

ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist

added 15 packages, and audited 1210 packages in 9s

181 packages are looking for funding
  run `npm fund` for details

4 vulnerabilities (2 high, 2 critical)

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

> test:typings:test
> npm run ts && npm run clean:tests && node tests/typings/setup --ts=3.9 && npm run ts && npm run clean:tests


> ts
> run-p ts:*


> ts:webdriverio
> cd tests/typings/webdriverio && npx tsc --skipLibCheck


> ts:devtools
> cd tests/typings/devtools && npx tsc --skipLibCheck


> ts:jasmine
> cd tests/typings/jasmine && npx tsc --skipLibCheck


> ts:cucumber
> cd tests/typings/cucumber && npx tsc --skipLibCheck


> ts:webdriver
> cd tests/typings/webdriver && npx tsc --skipLibCheck


> ts:mocha
> cd tests/typings/mocha && npx tsc --skipLibCheck


> clean:tests
> rimraf tests/typings/**/node_modules tests/typings/**/dist ./coverage

ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist
ln: Source file does not exist

> ts
> run-p ts:*


> ts:webdriver
> cd tests/typings/webdriver && npx tsc --skipLibCheck


> ts:devtools
> cd tests/typings/devtools && npx tsc --skipLibCheck


> ts:mocha
> cd tests/typings/mocha && npx tsc --skipLibCheck


> ts:cucumber
> cd tests/typings/cucumber && npx tsc --skipLibCheck


> ts:webdriverio
> cd tests/typings/webdriverio && npx tsc --skipLibCheck


> ts:jasmine
> cd tests/typings/jasmine && npx tsc --skipLibCheck


> clean:tests
> rimraf tests/typings/**/node_modules tests/typings/**/dist ./coverage


> test:typings:recover
> rimraf ./node_modules/.bin/tsc && ln -s ../typescript/bin/tsc node_modules/.bin && npx rimraf @tsd/typescript

'ln' is not recognized as an internal or external command,
operable program or batch file.
ERROR: "test:typings:recover" exited with 1.
ERROR: "test:typings" exited with 1.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Interesting, linking works when setting up our project in Windows here on GitHub. Not sure why `ln` is only supported there. Maybe we have to make an additional binary a dependency for Windows user. Anyway, I'ld suggest to replace this command in our `package.json` which something like `rimraf` which works cross operating systems.

Any contributions that resolves the bug are highly appreciated. Please take a look into our [contribution guidelines](https://webdriver.io/docs/contribute) and let us know if you have any questions. Cheers!

**Comment by christian-bromann** [comment]
Sorry for my late response.

> 1. ln: Source file does not exist

We can ignore this.

> 2. ln: Source file does not exist

Same.

> 3. 'ln' is not recognized as an internal or external command,

Adding `shx` seems like a good solution to me.

> 4. should process file name with []

This issue should be resolved now.

**Comment by RossVertizan** [comment]
@christian-bromann I am ready to go with the shx fix, but before I submit the PR thought I would have a look at the smoke tests.  The smoke tests have a lot of deliberate errors, so it's not entirely clear which are real and which are expected errors.  The output from the smoke tests is in the attached file.

[smoke_tests_windows.md](https://github.com/webdriverio/webdriverio/files/10031865/smoke_tests_windows.md)

I assume your known issue is the SevereServiceError: ups right at the start?

I am not aware of anything else "wonky" on Windows.  If there is I'll file an issue :)

**Comment by christian-bromann** [comment]
Thanks! Yeah, we use smoke tests to check whether the testrunner is failing in expected situations. If they pass then you can ignore these errors being shown.

**Comment by RossVertizan** [comment]
So everything is OK?

**Comment by RossVertizan** [comment]
OK, I'll submit the PR for the shx fix.

**Comment by RossVertizan** [comment]
Fixed by PR #9200
